framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/runtimes/AbstractLicenseOnlyDownloadExecutor.java

```diff
@@ -115,27 +118,28 @@ public WorkflowResponse executeDownload(DownloadSingleRuntimeRequest req) {
 	
 	private void initiateDownload(DownloadSingleRuntimeRequest req, DownloadRuntime dlrt, 
 			IRuntimeInstaller installer, File uniqueLoc, File downloads) {
-		String jobId = model.getJobManager().generateJobId();
 		String jobName = "Download runtime: " + dlrt.getId();
-		final IJob job = new SimpleJob(jobName, jobId);
-		model.getJobManager().addJob(job);
-		new Thread(jobName) {
-			public void run() {
+		
+		IStatusRunnableWithProgress task = new IStatusRunnableWithProgress() {
+			
+			@Override
+			public IStatus run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
 				// TODO, implement a progress monitor that can update progress
 				TaskModel tm2 = createDownloadTaskModel(req);
 				IStatus ret = installer.installRuntime(dlrt, uniqueLoc.getAbsolutePath(), downloads.getAbsolutePath(), 
-						true, tm2, new NullProgressMonitor());
+						true, tm2, monitor);
 				if( !ret.isOK()) {
-					model.getJobManager().removeJob(job, ret);
-					return;
+					return ret;
 				}
 				
 				// Now it's downloaded, but, we should now maybe install it? Or add it as a server?
 				String newHome = (String)tm2.getObject(IDownloadRuntimeWorkflowConstants.UNZIPPED_SERVER_HOME_DIRECTORY);
 				IStatus complete = createServer(dlrt, newHome);
```

Comment:
given that this not only downloads the runtime but also creates the server, I'd reflect this somehow. Either in method name, class name or at least in javadoc